packages/content/src/content.ts

```diff
@@ -19,7 +19,12 @@ export function getContent(
   if (requestOptions && requestOptions.isPortal) {
     return getContentFromPortal(id, requestOptions);
   } else {
-    return getContentFromHub(id, requestOptions);
+    return getContentFromHub(id, requestOptions).catch(() => {
+      // TODO: inspect error?
+      // dataset is not in index (i.e. might be a private item)
+      // try fetching from portal instead
+      return getContentFromPortal(id, requestOptions);
```

Comment:
I don't love the fact that we use `portal` to refer both to enterprise sites and to the portal API here (and everywhere else in the app). Maybe we could consider calling this function `getContentFromPortalApi` or something. Anyway, your call.

Reply:
Fair point. I would suggest we also rename `getContentFromHubApi` as well then.